core/src/main/java/hudson/model/FreeStyleProject.java

```diff
@@ -72,5 +74,25 @@ public String getDisplayName() {
         public FreeStyleProject newInstance(ItemGroup parent, String name) {
             return new FreeStyleProject(parent,name);
         }
+
+        @Override
+        public String getDescription() {
+            return Messages.FreeStyleProject_Description();
+        }
+
+        @Override
+        public String getCategoryId() {
+            return StandaloneProjectsCategory.ID;
+        }
+
+        @Override
+        public String getIconFilePathPattern() {
+            if (Jenkins.RESOURCE_PATH.isEmpty()) {
+                return "images/:size/freestyleproject.png";
+            } else {
+                return Jenkins.RESOURCE_PATH.substring(1) + "/images/:size/freestyleproject.png";
```

Comment:
Also advisable for the Javadoc for this method to suggest prepending `RESOURCE_PATH` even for plugin-served images; `Plugin.doDynamic` honors it. (The current downstream PRs are not doing so. Not a major concern since users presumably are not hitting the _New Item_ screen constantly; we mostly care about enforcing `RESOURCE_PATH` usage on things like sidepanel links where it has a big effect on page load performance.)